crates/router/src/core/payments/operations/payment_response.rs

```diff
@@ -615,70 +616,113 @@ impl<F: Clone> PostUpdateTracker<F, PaymentData<F>, types::SdkPaymentsSessionUpd
         _payment_id: &api::PaymentIdType,
-        payment_data: PaymentData<F>,
-        _router_data: types::RouterData<
+        mut payment_data: PaymentData<F>,
+        router_data: types::RouterData<
             F,
             types::SdkPaymentsSessionUpdateData,
             types::PaymentsResponseData,
         >,
-        _key_store: &domain::MerchantKeyStore,
-        _storage_scheme: enums::MerchantStorageScheme,
+        key_store: &domain::MerchantKeyStore,
+        storage_scheme: enums::MerchantStorageScheme,
         _locale: &Option<String>,
         #[cfg(feature = "dynamic_routing")] _routable_connector: Vec<RoutableConnectorChoice>,
         #[cfg(feature = "dynamic_routing")] _business_profile: &domain::Profile,
     ) -> RouterResult<PaymentData<F>>
     where
         F: 'b + Send,
     {
-        // let session_update_details =
-        //     payment_data
-        //         .payment_intent
-        //         .tax_details
-        //         .clone()
-        //         .ok_or_else(|| {
-        //             report!(errors::ApiErrorResponse::InternalServerError)
-        //                 .attach_printable("missing tax_details in payment_intent")
-        //         })?;
-
-        // let pmt_amount = session_update_details
-        //     .pmt
-        //     .clone()
-        //     .map(|pmt| pmt.order_tax_amount)
-        //     .ok_or(errors::ApiErrorResponse::InternalServerError)
-        //     .attach_printable("Missing tax_details.order_tax_amount")?;
-
-        // let total_amount = MinorUnit::from(payment_data.amount) + pmt_amount;
-
-        // // if connector_ call successful -> payment_intent.amount update
-        // match router_data.response.clone() {
-        //     Err(_) => (None, None),
-        //     Ok(types::PaymentsResponseData::SessionUpdateResponse { status }) => {
-        //         if status == SessionUpdateStatus::Success {
-        //             (
-        //                 Some(
-        //                     storage::PaymentAttemptUpdate::IncrementalAuthorizationAmountUpdate {
-        //                         amount: total_amount,
-        //                         amount_capturable: total_amount,
-        //                     },
-        //                 ),
-        //                 Some(
-        //                     storage::PaymentIntentUpdate::IncrementalAuthorizationAmountUpdate {
-        //                         amount: pmt_amount,
-        //                     },
-        //                 ),
-        //             )
-        //         } else {
-        //             (None, None)
-        //         }
-        //     }
-        //     _ => Err(errors::ApiErrorResponse::InternalServerError)
-        //         .attach_printable("unexpected response in session_update flow")?,
-        // };
-
-        // let _shipping_address = payment_data.address.get_shipping();
-        // let _amount = payment_data.amount;
+        let connector = payment_data
```

Comment:
we need connector only in case its an error, then why are we throwing the error here?